libsolidity/formal/SMTEncoder.cpp

```diff
@@ -579,6 +579,28 @@ void SMTEncoder::endVisit(BinaryOperation const& _op)
 		);
 }
 
+bool SMTEncoder::visit(Conditional const& _op)
+{
+	_op.condition().accept(*this);
+
+	auto indicesEndTrue = visitBranch(&_op.trueExpression(), expr(_op.condition()));
+	auto touchedVars = touchedVariables(_op.trueExpression());
+	decltype(indicesEndTrue) indicesEndFalse;
```

Comment:
For `IfStatement` it's done this way because it might not have a false statement. Since a conditional must have one, you could write directly
`auto indicesEndFalse = visitBranch(&_op.falseExpression(), !expr(_op.condition()));`
and remove the line above.